### DISCUSSION TITLE
Help installing?

### DISCUSSION THREAD

**Discussion by JoeyLambs** [Code-Forge-Net/Remix-Dev-Tools#149] [General]
I'm trying to install this plugin to help debug a hydration issue in mt Shopify Hydrogen site.

Here's my vite config

I'm getting an error
```
ReferenceError: require is not defined
    at C:/_dev/cbc/hydrogen/node_modules/date-fns/add/index.js:3:30
    at /node_modules/remix-development-tools/dist/client.js?v=d08249d1:62:32
    at /app/root.tsx:1:110
    ~
   ```
   Usually I think this error means I need to install types, but I tried `npm install @types/remix-development-tools` and there was nothing there, so I'm lost. I could use some help setting this up, TIA.

```
import {vitePlugin as remix} from '@remix-run/dev';
import {hydrogen} from '@shopify/hydrogen/vite';
import {oxygen} from '@shopify/mini-oxygen/vite';
import tailwindcss from '@tailwindcss/vite';
import {remixDevTools} from 'remix-development-tools';
import {defineConfig} from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  plugins: [
    tailwindcss(),
    hydrogen(),
    oxygen(),
    remixDevTools(),
    remix({
      presets: [hydrogen.preset()],
      future: {
        v3_fetcherPersist: true,
        v3_relativeSplatPath: true,
        v3_throwAbortReason: true,
      },
    }),
    tsconfigPaths(),
  ],
  build: {
    // Allow a strict Content-Security-Policy
    // withtout inlining assets as base64:
    assetsInlineLimit: 0,
  },
  ssr: {
    optimizeDeps: {
      /**
       * Include dependencies here if they throw CJS<>ESM errors.
       * For example, for the following error:
       *
       * > ReferenceError: module is not defined
       * >   at /Users/.../node_modules/example-dep/index.js:1:1
       *
       * Include 'example-dep' in the array below.
       * @see https://vitejs.dev/config/dep-optimization-options
       */
      include: ['react-slick', 'react-responsive-carousel'],
    },
  },
});

```

**Comment by AlemTuzlak** [comment]
is your project CJS by any chance?

**Reply by JoeyLambs** [reply]
not 100% sure, since I don't recall actively choosing one or the other, but look like ems based on my package.json
![image](https://github.com/user-attachments/assets/5bbc331a-a884-4df0-b3c1-3ba64d8037ec)

again, not sure, though

**Reply by AlemTuzlak** [reply]
could you post your tsconfig.json as well, that will go a long way, considering that RDT is ESM only requires don't exist in the distributed version

**Reply by JoeyLambs** [reply]
```
{
  "include": ["./**/*.d.ts", "./**/*.ts", "./**/*.tsx"],
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2022"],
    "isolatedModules": true,
    "esModuleInterop": true,
    "jsx": "react-jsx",
    "moduleResolution": "Bundler",
    "resolveJsonModule": true,
    "module": "ES2022",
    "target": "ES2022",
    "strict": true,
    "allowJs": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true,
    "baseUrl": ".",
    "types": ["@shopify/oxygen-workers-types"],
    "paths": {
      "~/*": ["app/*"]
    },
    "noEmit": true
  }
}

**Reply by AlemTuzlak** [reply]
interesting, looks good, could you maybe create a minimum repro? I wonder if its caused due to hydrogen/oxygen. I've never seen this happen before, first thing that comes to mind is adding it to no-external

**Reply by JoeyLambs** [reply]
Will take some time to set up the Repro, I'll get back to you. Thanks for your help!

**Reply by JoeyLambs** [reply]
https://github.com/Nicolis1/RemixDevToolRepro
Didn't take as long as I expected.

I ran 
`npm create @shopify/hydrogen@latest -- --quickstart`

then 
`npm install remix-development-tools -D`

'npm run dev'

'''
MiniOxygen couldn't load your app's entry point.
ReferenceError: require is not defined
    at C:/_dev/cbc/hydrogen-quickstart/node_modules/date-fns/add/index.js:3:30
    at /node_modules/remix-development-tools/dist/client.js?v=9e1f414b:62:32
    at /app/root.jsx:1:110
    '''

**Reply by AlemTuzlak** [reply]
It seems there's an issue with hydrogen/oxygen & rdt, i'll try to figure out what's going on

**Reply by AlemTuzlak** [reply]
I think I figured out the issue, I'll investigate more and get back to you with a fix

**Reply by JoeyLambs** [reply]
You're the best! Thank you very much

**Reply by AlemTuzlak** [reply]
i'll release a new version tomorrow with a dedicated how to page on the docs on how to fix this. Figured it out.

**Reply by JoeyLambs** [reply]
It's working! Thank you so much!

**Comment by AlemTuzlak** [comment]
Install the latest version and follow this guide:
https://remix-development-tools.fly.dev/docs/main/hydrogen

PS:
Thank you for sponsoring me!!!
